src/convert.ts

```diff
@@ -228,70 +253,76 @@ export function convertShare(json: unknown): ShareResult {
+          let knownContainerUniqueIds = ["/"];
+          if (Array.isArray(v7Member.parents) && v7Member.parents.length > 0) {
+            knownContainerUniqueIds = v7Member.parents
+              .map((parent: string) => {
+                // Convert v7 user added data group id
+                if (parent === "Root Group/User-Added Data") {
+                  return "__User-Added_Data__";
+                }
+                // Convert v7 root group id
+                if (parent === "Root Group") {
+                  return "/";
+                }
+
+                // Replace v7 autoID
+                return parent;
+              })
+              .filter((parent) => typeof parent !== "undefined") as string[];
+          }
+
+          // Use model id if it is defined, otherwise use key from members object
+          let newId =
+            typeof v7Member.id === "string" && v7Member.id !== ""
+              ? v7Member.id
+              : v7id;
+
+          // Replace User Added Data group id
+          if (v7id === "Root Group/User-Added Data") {
+            newId = "__User-Added_Data__";
+          }
+
+          // For some reason user added data doesn't have "__User-Added_Data__" in v8 autoIDs, whereas v7 autoIDs start with `Root Group/User-Added Data`,
+          // so remove all mentions of Root Group/User-Added Data from v7 autoIDs
+          newId = newId.replace("Root Group/User-Added Data", "");
+
+          // Only add to workbenchIds if NOT converting User Added Data
+          if (v7Member.isEnabled && !convertUserAdded) {
+            workbenchIds.push({
+              id: newId,
+              index:
+                typeof v7Member.nowViewingIndex === "number"
+                  ? v7Member.nowViewingIndex
+                  : undefined,
+            });
+          }
+
+          // Only convert user added data if convertUserAdded
+          if (
+            convertUserAdded ||
+            (v7id !== "Root Group/User-Added Data" &&
+              !knownContainerUniqueIds.includes("__User-Added_Data__"))
+          ) {
+            const result = convertMember(v7Member, { partial: true });
+            messages.push(...result.messages);
+            convertedMembers[newId] = {
+              ...result.member,
+              knownContainerUniqueIds,
+            };
+          }
+          return convertedMembers;
         }
-
-        // Only convert user added data if convertUserAdded
-        if (
-          convertUserAdded ||
-          (id !== "Root Group/User-Added Data" &&
-            !knownContainerUniqueIds.includes("__User-Added_Data__"))
-        ) {
-          const result = convertMember(v7Member, { partial: true });
-          messages.push(...result.messages);
-          convertedMembers[newId] = {
-            ...result.member,
-            knownContainerUniqueIds,
-          };
-        }
-        return convertedMembers;
-      }
-    }, {});
```

Comment:
This has done a really annoying diff. 
It is way less exciting than it looks:
![image](https://user-images.githubusercontent.com/6187649/102171100-d9f5d180-3ee9-11eb-903e-958e9aaa72d8.png)